<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>testDartComputingByTS</class>
 <widget class="QWidget" name="testDartComputingByTS">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>标定计算器</string>
  </property>
  <widget class="QLabel" name="testProcessLabel">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>410</y>
     <width>286</width>
     <height>48</height>
    </rect>
   </property>
   <property name="text">
    <string>标定过程：step1:测(m镖·g)和所有坐标。计算得到x和h
         step2:T测拉到F₀,发射后计算出∫F₀·dx拉
         step3:ΔT测变化,发射后计算出k1·x拉</string>
   </property>
  </widget>
  <widget class="QLabel" name="deltaTensionFormulaLabel">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>406</y>
     <width>446</width>
     <height>48</height>
    </rect>
   </property>
   <property name="text">
    <string>∫F₀·dx拉 = (m镖 + m发射台)·g·x²/{ 4·cos²(θ)·[x·tan(θ)  - h]}    (ΔT测 = 0)

 k1·x拉 = {(m镖 + m发射台)·g·x²/{ 4·cos²(θ)·[x·tan(θ)  - h]} - ∫F₀·dx拉} / ΔT测</string>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>190</y>
     <width>1181</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Orientation::Horizontal</enum>
   </property>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>12</y>
     <width>1178</width>
     <height>37</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QPushButton" name="testDartPushButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>第零步:标定飞镖(F10)</string>
      </property>
      <property name="shortcut">
       <string>F10</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_6">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>180</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="ConnectUartPushButton">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>第一步:连接串口(F1)</string>
      </property>
      <property name="shortcut">
       <string>F1</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_4">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>99</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="yawAimingPushButton">
      <property name="text">
       <string>第二步:yaw轴瞄准(F2)</string>
      </property>
      <property name="shortcut">
       <string>F2</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_5">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>168</width>
        <height>25</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="dartsParasComputingPushButton">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>第三步:飞镖参数计算(F3)</string>
      </property>
      <property name="shortcut">
       <string>F3</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>60</y>
     <width>1181</width>
     <height>121</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="7">
     <widget class="QTextEdit" name="leadRightFrontCoordTextEdit"/>
    </item>
    <item row="1" column="6">
     <widget class="QTextEdit" name="leadLeftFrontCoordTextEdit"/>
    </item>
    <item row="1" column="9">
     <widget class="QTextEdit" name="leadRightBackCoordTextEdit"/>
    </item>
    <item row="0" column="7">
     <widget class="QLabel" name="setaLabel_9">
      <property name="text">
       <string>12:导轨右上</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QTextEdit" name="rackLeftBackCoordTextEdit"/>
    </item>
    <item row="0" column="10">
     <widget class="QLabel" name="setaLabel_10">
      <property name="text">
       <string>14:飞镖发射点</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QTextEdit" name="targetCoordTextEdit"/>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="setaLabel_3">
      <property name="text">
       <string>6:架左后方</string>
      </property>
     </widget>
    </item>
    <item row="1" column="4">
     <widget class="QTextEdit" name="rackRightFrontTextEdit"/>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="setaLabel_4">
      <property name="text">
       <string>7:导轨左下</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QLabel" name="setaLabel_2">
      <property name="text">
       <string>1:目标坐标</string>
      </property>
     </widget>
    </item>
    <item row="1" column="10">
     <widget class="QTextEdit" name="leadDartShootCoordTextEdit"/>
    </item>
    <item row="0" column="5">
     <widget class="QLabel" name="setaLabel_7">
      <property name="text">
       <string>10:架左前方</string>
      </property>
     </widget>
    </item>
    <item row="1" column="5">
     <widget class="QTextEdit" name="rackLeftFrontTextEdit"/>
    </item>
    <item row="0" column="4">
     <widget class="QLabel" name="setaLabel_6">
      <property name="text">
       <string>9:架右前方</string>
      </property>
     </widget>
    </item>
    <item row="0" column="6">
     <widget class="QLabel" name="setaLabel_8">
      <property name="text">
       <string>11:导轨左上</string>
      </property>
     </widget>
    </item>
    <item row="0" column="9">
     <widget class="QLabel" name="setaLabel_11">
      <property name="text">
       <string>13:导轨右下</string>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QTextEdit" name="rackRightBackTextEdit"/>
    </item>
    <item row="0" column="3">
     <widget class="QLabel" name="setaLabel_5">
      <property name="text">
       <string>8:架右后方</string>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QTextEdit" name="leadLeftBackCoordTextEdit"/>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>290</y>
     <width>721</width>
     <height>60</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QLabel" name="setaLabel">
      <property name="text">
       <string>θ:当前发射轨道pitch角(度)</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="xLabel">
      <property name="text">
       <string>x:发射点到目标的水平距离(米)</string>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QLabel" name="hLabel">
      <property name="text">
       <string>h:发射点到目标的竖直距离(米)</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLineEdit" name="setaLineEdit">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="xLineEdit">
      <property name="text">
       <string/>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QLineEdit" name="hLineEdit">
      <property name="text">
       <string/>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="3">
     <widget class="QPushButton" name="computeXandHPushButton">
      <property name="text">
       <string>计算θ,x,h</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>486</y>
     <width>1181</width>
     <height>186</height>
    </rect>
   </property>
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="1">
     <widget class="QLabel" name="mLauncherPlusGLabelInput">
      <property name="text">
       <string>m发射台·g</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="mLauncherPlusGLineEditInput"/>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="mdart1PlusGlabelInput_2">
      <property name="text">
       <string>m镖·g(g·N/kg)</string>
      </property>
     </widget>
    </item>
    <item row="2" column="5">
     <widget class="QLabel" name="integralOfF0PlusDxtensionLabelOutput">
      <property name="text">
       <string>标定得到的∫F₀·dx拉</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLineEdit" name="mdart1PlusGLineEditInput"/>
    </item>
    <item row="3" column="4">
     <widget class="QPushButton" name="computeTall4PushButton">
      <property name="text">
       <string>计算 ∫F₀·dx拉(F5)</string>
      </property>
      <property name="shortcut">
       <string>F5</string>
      </property>
     </widget>
    </item>
    <item row="3" column="5">
     <widget class="QLineEdit" name="integralOfF0PlusDxtensionLineEditOutput">
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="f0LabelInput">
      <property name="text">
       <string>自己设置的F₀</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QLabel" name="mdart1PlusGlabelInput">
      <property name="text">
       <string>m镖·g(g·N/kg)</string>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QLabel" name="integralOfF0PlusDxtensionLabelInput">
      <property name="text">
       <string>∫F₀·dx拉</string>
      </property>
     </widget>
    </item>
    <item row="4" column="3">
     <widget class="QLabel" name="Tall1LabelInput">
      <property name="text">
       <string>起始的 T总 = F₀ + ΔT测</string>
      </property>
     </widget>
    </item>
    <item row="4" column="4">
     <widget class="QPushButton" name="copyTall4PushButton">
      <property name="text">
       <string>复制 ∫F₀·dx拉(F6)</string>
      </property>
      <property name="shortcut">
       <string>F6</string>
      </property>
     </widget>
    </item>
    <item row="4" column="5">
     <widget class="QLabel" name="k1PlusXtensionLabelInput">
      <property name="text">
       <string>标定得到的k1·x拉</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLineEdit" name="f0LineEditInput"/>
    </item>
    <item row="5" column="1">
     <widget class="QLineEdit" name="mdart2PlusGLineEditInput"/>
    </item>
    <item row="5" column="2">
     <widget class="QLineEdit" name="integralOfF0PlusDxtensionLineEditInput">
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="3">
     <widget class="QLineEdit" name="Tall1LineEditInput">
      <property name="text">
       <string/>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="5" column="4">
     <widget class="QPushButton" name="computeK1PlusXtensionPushButton">
      <property name="text">
       <string>计算 k1·x拉(F7)</string>
      </property>
      <property name="shortcut">
       <string>F7</string>
      </property>
     </widget>
    </item>
    <item row="5" column="5">
     <widget class="QLineEdit" name="k1PlusXtensionLineEditInput">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>computeXandHPushButton</tabstop>
  <tabstop>f0LineEditInput</tabstop>
  <tabstop>computeK1PlusXtensionPushButton</tabstop>
  <tabstop>setaLineEdit</tabstop>
  <tabstop>mLauncherPlusGLineEditInput</tabstop>
  <tabstop>mdart1PlusGLineEditInput</tabstop>
  <tabstop>computeTall4PushButton</tabstop>
  <tabstop>copyTall4PushButton</tabstop>
  <tabstop>Tall1LineEditInput</tabstop>
  <tabstop>integralOfF0PlusDxtensionLineEditInput</tabstop>
  <tabstop>mdart2PlusGLineEditInput</tabstop>
  <tabstop>dartsParasComputingPushButton</tabstop>
  <tabstop>ConnectUartPushButton</tabstop>
  <tabstop>testDartPushButton</tabstop>
  <tabstop>integralOfF0PlusDxtensionLineEditOutput</tabstop>
  <tabstop>hLineEdit</tabstop>
  <tabstop>k1PlusXtensionLineEditInput</tabstop>
  <tabstop>xLineEdit</tabstop>
  <tabstop>yawAimingPushButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
